datastore/emulator.go

```diff
@@ -0,0 +1,115 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package datastore
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	log "github.com/sirupsen/logrus"
+	"time"
+
+	//"net/http"
+	"os"
+	"os/exec"
+	"syscall"
+)
+
+// Fake projectID set by the emulator
+var fakeProjectID = "testapp"
+
+// Port for emulator to use
+var emulatorPort = 30801
+
+// Datastore emulator for use in tests
+//
+// Background: this was devised due to aetest package no longer being available for 1.12+ runtimes as dev_appserver.py
+// cannot run runtimes past go111, see: https://github.com/googleapis/google-cloud-go/issues/644
+//
+//   Usage:
+//
+//   testCtx, testCtxCancel := context.WithCancel(context.Background())
+//   err := datastore.Emulator(testCtx)
+//   testCtxCancel()
+//
+func Emulator(ctx context.Context) error {
+
+	// Set log formatter to process newlines since emulator gives a lot of output
+	log.SetFormatter(&log.TextFormatter{DisableQuote: true})
```

Comment:
This changes the global logging state, probably not what you want. logrus can create child loggers with different formats which might be good here.

or move this SetFormatter line to main().

Reply:
Good point. I dug quite a bit through the docs, but couldn't find specifics on child loggers, can you point at an example somewhere?

Reply:
https://godoc.org/github.com/sirupsen/logrus#New

Reply:
Fixed it via `log.WithContext()` PTAL